### DISCUSSION TITLE
Use AbilityBuilder or createMongoAbility??

### DISCUSSION THREAD

**Discussion by douglasg14b** [stalniy/casl#898] [Q&A]
I'm not finding the clarification I need to know why I would use one over the other here.

I'm also confused why using `createMongoAbility` retsult in eslint errors when deconstructed here. Which is also mentioned here https://github.com/stalniy/casl/discussions/835

```ts

type Actions = 'create' | 'read' | 'update' | 'delete';
type Subjects = 'Article' | 'Comment' | 'User';

// eslint@typescript-eslint/unbound-method error
const { can, cannot, build } = createMongoAbility<[Actions, Subjects]>();

// vs 

type AppAbility = MongoAbility<[Actions, Subjects]>;
const { can, cannot, build } = new AbilityBuilder<AppAbility>(createMongoAbility);

```

**Comment by stalniy** [comment]
Hello,

`createMongoAbility` creates an Ability instance from rules (array of POJO) and `AbilityBuilder` is declarative builder to define abilities (so it’s just abilities sugar)
